presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -301,6 +301,17 @@ public static long timestampAtTimeZone(@SqlType(StandardTypes.TIMESTAMP_WITH_TIM
         return packDateTimeWithZone(unpackMillisUtc(timestampWithTimeZone), getTimeZoneKeyForOffset(zoneOffsetMinutes));
     }
 
+    @ScalarFunction(value = "with_timezone")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE)
+    public static long withTimezone(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP) long timestamp, @SqlType("varchar(x)") Slice zoneId)
+    {
+        String zoneIdString =  zoneId.toStringUtf8();
+        DateTimeZone fromDateTimeZone = session.isLegacyTimestamp() ? getDateTimeZone(session.getTimeZoneKey()) : DateTimeZone.UTC;
+        DateTimeZone toDateTimeZone = getDateTimeZone(getTimeZoneKey(zoneIdString));
```

Comment:
Communication glitch :)
I mean something like:

```suggestion
        TimeZoneKey toTimeZoneKey = getTimeZoneKey(zoneId.toStringUtf8());
        DateTimeZone toDateTimeZone = getDateTimeZone(toTimeZoneKey);
```